app/controllers/spree/api/v2/storefront/wished_products_controller.rb

```diff
@@ -0,0 +1,66 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class WishedProductsController < ::Spree::Api::V2::BaseController
+
+          def create
+            spree_authorize! :create, Spree::WishedProduct
+            spree_authorize! :update, wishlist
+
+            wished_product = Spree::WishedProduct.new(wished_product_attributes)
+
+            if wishlist.include? params[:wished_product][:variant_id]
+              wished_product = wishlist.wished_products.detect {|wp| wp.variant_id == params[:wished_product][:variant_id].to_i }
+            else
+              wished_product.wishlist = wishlist
+              wished_product.save
+            end
+
+            wishlist.reload
+            if wished_product.persisted?
+              render_serialized_payload { serialize_resource(wished_product) }
+            else
+              render_error_payload(wished_product.errors.full_messages.join(', '))
```

Comment:
```suggestion
              render_error_payload(wished_product.errors.full_messages.to_sentence)
```

you can extract that into a separate method as it's used 2 times